core/src/main/java/org/pentaho/platform/genericfile/providers/repository/RepositoryFileProvider.java

```diff
@@ -0,0 +1,171 @@
+ * or from the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Lesser General Public License for more details.
+ *
+ *
+ * Copyright (c) 2023 Hitachi Vantara. All rights reserved.
+ *
+ */
+
+package org.pentaho.platform.genericfile.providers.repository;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import org.pentaho.platform.api.genericfile.IGenericFileProvider;
+import org.pentaho.platform.api.repository2.unified.IUnifiedRepository;
+import org.pentaho.platform.api.repository2.unified.webservices.RepositoryFileDto;
+import org.pentaho.platform.api.repository2.unified.webservices.RepositoryFileTreeDto;
+import org.pentaho.platform.engine.core.system.PentahoSessionHolder;
+import org.pentaho.platform.engine.core.system.PentahoSystem;
+import org.pentaho.platform.genericfile.messages.Messages;
+import org.pentaho.platform.genericfile.providers.repository.model.RepositoryFile;
+import org.pentaho.platform.genericfile.providers.repository.model.RepositoryFileTree;
+import org.pentaho.platform.genericfile.providers.repository.model.RepositoryFolder;
+import org.pentaho.platform.genericfile.providers.repository.model.RepositoryObject;
+import org.pentaho.platform.web.http.api.resources.services.FileService;
+
+import java.util.Date;
+
+import static org.pentaho.platform.util.RepositoryPathEncoder.encodeRepositoryPath;
+
+public class RepositoryFileProvider implements IGenericFileProvider<RepositoryFile> {
+  public static String REPOSITORY_PREFIX = "/";
+  private IUnifiedRepository unifiedRepository;
+
+  @Override public Class<RepositoryFile> getFileClass() {
+    return RepositoryFile.class;
+  }
+
+  private RepositoryFileTree tree;
+  public static final String TYPE = "repository";
+
+  public RepositoryFileProvider() {
+    unifiedRepository = PentahoSystem.get( IUnifiedRepository.class, PentahoSessionHolder.getSession() );
+  }
+
+  @Override
+  public String getName() {
+    return Messages.getString( "GenericFileRepository.REPOSITORY_FOLDER_DISPLAY" );
+  }
+
+  @Override
+  public String getType() {
+    return TYPE;
+  }
+
+  @Override
+  public boolean createFolder( String path ) {
+    try {
+      FileService fileService = new FileService();
+
+      boolean success = fileService.doCreateDirSafe( encodeRepositoryPath( path ) );
+      if ( success ) {
+        clearFolderCache();
+      }
+
+      return success;
+    } catch ( FileService.InvalidNameException e ) {
+      return false;
+    }
+  }
+
+  @Override
+  public RepositoryFileTree getFolders( Integer depth ) {
+    if ( tree != null ) {
+      return tree;
+    }
+
+    FileService fileService = new FileService();
```

Comment:
Don't have to address now, but shouldn't this be instantainted via spring?
Either through spring annotations or `PentahoSystem.get()`

Reply:
Good point. 

Initially, we had to create an instance each time, because the provider was as singleton. Now, the provider is session based, and so we could easily initialize it at construction.

However, regarding Spring or Pentaho system. This is a class having no interface. The top Pentaho container has no bean for this class. All uses of it create an instance of it directly.

I believe we should allow receiving this via constructor, as well as to have a default constructor which creates one instance of it.

It would help unit tests.

Regarding Spring configuration, given current practice, I don't see much value.